synapse/handlers/message.py

```diff
@@ -1016,9 +1055,12 @@ async def create_new_client_event(
             if already_exists:
                 raise SynapseError(400, "Can't send same reaction twice")
 
-        logger.debug("Created event %s", event.event_id)
-
-        return event, context
+        # Don't attempt to start a thread if the parent event is a relation.
+        elif relation_type == RelationTypes.THREAD:
+            if await self.store.event_has_relations(relates_to):
+                raise SynapseError(
+                    400, "Cannot start threads from an event with a relation"
+                )
```

Comment:
> Does this mean that nested threading is not possible?

Yes, it means you can not nest threads (and cannot start threads off of, e.g. a reaction). Is there a way I can make this clearer? Maybe the method should be `event_is_relation` instead of `event_has_relations`. It is checking if the event you're trying to create a relation to also has a relation.

Reply:
I feel like `event_has_parent` or similar would clarify this particular case a lot (though it probably wants the word `relation` in there ... `event_has_relation_parent`? hmm)
Mind you, relations are meant to be a bit abstract and don't necessarily mean parent/child, but the current uses for them do kind of seem parent/child-like (and the parent has to have been born first, since the child refers to it by ID...).
Naming is hard.

Reply:
(though I will note that it's sad that nested threading will not be an option here, I'll have to trust that this has been through MSCs and people are happy that there is a workaround for nested threading in the future)

Reply:
> (though I will note that it's sad that nested threading will not be an option here, I'll have to trust that this has been through MSCs and people are happy that there is a workaround for nested threading in the future)

There's various reasons given in the MSC, this conversation belongs on there. 😄